Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -1436,58 +1437,35 @@ void NetworkConnectionToWebProcess::installMockContentFilter(WebCore::MockConten
 }
 #endif
 
+
 #if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
-void NetworkConnectionToWebProcess::logOnBehalfOfWebContent(const String& logChannel, const String& logString, uint8_t logType)
+static CString validatedASCIIString(const String& string, unsigned maxLength)
 {
-    WTFLogChannel* channel = getLogChannel(logChannel);
-    if (!channel)
-        channel = WebCore::getLogChannel(logChannel);
-    if (!channel)
-        channel = WTF::getLogChannel(logChannel);
-
-    CString string = logString.ascii();
-
-    // Truncate long log strings and only log printable characters
-    constexpr unsigned maxLength = 256;
-    auto stringLength = string.length();
-
-    if (stringLength > maxLength) {
-        string.mutableData()[maxLength] = 0;
-        stringLength = maxLength;
-    }
+    String shortString = string.substring(0, maxLength);
+    CString validatedString = shortString.ascii();
+    auto stringLength = validatedString.length();
 
     for (unsigned i = 0; i < stringLength; i++) {
-        char c = string.data()[i];
+        char c = validatedString.data()[i];
         if (c >= 32 && c <= 126)
             continue;
-
-        string.mutableData()[i] = ' ';
+        validatedString.mutableData()[i] = ' ';
     }
+    return validatedString;
+}
 
-#define LOGFORMATSTRING "WebContent (%llu): %s"
+void NetworkConnectionToWebProcess::logOnBehalfOfWebContent(const String& logChannel, const String& logCategory, const String& logString, uint8_t logType, int32_t pid)
+{
+    // Truncate long strings and only accept printable characters
+    CString string = validatedASCIIString(logString, 256);
```

Comment:
What's the longest string we log today? Probably a limit like 2X that would be appropriate.

Reply:
> What's the longest string we log today? Probably a limit like 2X that would be appropriate.

That is a good point! I have adjusted the limit accordingly.

Thanks for reviewing!